### ISSUE TITLE
--no-optional flag has no effect

### ISSUE THREAD

**Issue by nickpape-msft** [pnpm/pnpm#1014]
### Code to reproduce the issue:

Create a package JSON:
```
{
  "name": "optional",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "optionalDependencies": {
    "colors": "*"
  },
  "author": "",
  "license": "ISC"
}
```

Run `pnpm install --no-optional`

### Expected behavior:

The `colors` package is not installed or included in shrinkwrap.

### Actual behavior:

`Colors` is installed and included in shrinkwrap.

### Additional information:

It seems like the work to support optional dependencies has already been done: https://github.com/pnpm/pnpm/issues/460

However, looking through the code it looks like the command-line flag was never hooked up.

 - `node -v` prints: 8.9.4
 - Windows, OS X, or Linux?: Windows

**Comment by zkochan** [comment]
fixed in [v1.31.0](https://github.com/pnpm/pnpm/releases/tag/v1.31.0)

**Comment by nickpape-msft** [comment]
Thanks Zoltan!
